### PR TITLE
Add SuccessfulResult static property

### PR DESCRIPTION
**Pull request by MikyM**
This PR adds a `SuccessfulResult` static property to `Result` much like `ValueTask`'s / `Task`'s `CompletedTask` property with a cached boxed instance of a successful result.

### CODE REVIEW

**Review by JeremyEspresso — APPROVED** [review]

File: Remora.Results/Result.cs [hunk 1]
```diff
@@ -32,6 +32,17 @@ namespace Remora.Results;
...
+    /// <summary>
+    /// Gets a successful result.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This property returns a result whose <see cref="Error"/> property is set to null, therefore <see cref="IsSuccess"/> returns true.
+    /// The <see cref="Inner"/> property is also set to null.
+    /// </para>
+    /// </remarks>
+    public static IResult SuccessfulResult { get; } = FromSuccess();
```
Comment (Nihlus) [inline Remora.Results/Result.cs]: I'd shorten this to `Success` since it's a static property and it'll always be called with a reference to the `Result` type.

Consider adding a matching property to the typed result struct as well for symmetry.

File: Remora.Results/Result.cs [hunk 2]
```diff
@@ -32,6 +32,17 @@ namespace Remora.Results;
 [PublicAPI]
 public readonly struct Result : IResult
 {
+    /// <summary>
+    /// Gets a successful result.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This property returns a result whose <see cref="Error"/> property is set to null, therefore <see cref="IsSuccess"/> returns true.
```
Comment (Nihlus) [inline Remora.Results/Result.cs]: You can remove this paragraph - these are implementation details and redundant information for this particular property.

File: Tests/Remora.Results.Tests/ResultTests.cs [hunk 3]
```diff
@@ -389,4 +389,41 @@ public void CreatesAnUnsuccessfulResultFromAnExceptionInstance()
...
+        {
+            var successful = Result.SuccessfulResult;
+            Assert.True(successful.IsSuccess);
+        }
+
+        /// <summary>
+        /// Tests whether <see cref="Result.SuccessfulResult"/> returns a result whose inner property is null.
+        /// </summary>
+        [Fact]
+        public void ReturnsAResultWhoseInnerPropertyIsNull()
```
Comment (Nihlus) [inline Tests/Remora.Results.Tests/ResultTests.cs]: You can remove this test - it's redundant since `IsSuccess` has static checking associated with its state, which you test above.

File: Remora.Results/Result.cs (hunk 1 above)
Comment (Nihlus) [inline Remora.Results/Result.cs]: I'd prefer the use of the full type here instead of the interface, since this will preclude the most common usage of this property - returning when the return type is known and not the interface. On the flip side, I'd support a separate property for exposing a boxed singleton - `BoxedSuccess`, perhaps?

**Review by Nihlus — CHANGES_REQUESTED** [review]

File: Remora.Results/Result.cs (hunk 1 above)
Comment (MikyM) [reply Remora.Results/Result.cs]: Shortened, added a matching property to the typed result aswell.

File: Remora.Results/Result.cs (hunk 2 above)
Comment (MikyM) [reply Remora.Results/Result.cs]: Removed the paragraph.

File: Tests/Remora.Results.Tests/ResultTests.cs (hunk 3 above)
Comment (MikyM) [reply Tests/Remora.Results.Tests/ResultTests.cs]: Test removed.

File: Remora.Results/Result.cs (hunk 1 above)
Comment (MikyM) [reply Remora.Results/Result.cs]: Property changed to use the full type and to be a computed one based on `FromSuccess()` (or `new(...)` for the typed result).

Added a `BoxedSuccess` property to both result types.

File: Remora.Results/Result.cs [hunk 4]
```diff
@@ -32,16 +32,18 @@ namespace Remora.Results;
 [PublicAPI]
 public readonly struct Result : IResult
 {
+    /// <summary>
+    /// Gets a successful result.
+    /// </summary>
+    public static Result Success => FromSuccess();
```
Comment (Nihlus) [inline Remora.Results/Result.cs]: You should cache this result as well, same as with the other one.

File: Remora.Results/Result.cs [hunk 5]
```diff
@@ -180,6 +182,19 @@ public static Result FromError<TEntity>(Result<TEntity> result)
 [PublicAPI]
 public readonly struct Result<TEntity> : IResult<TEntity>
 {
+    /// <summary>
+    /// Gets a successful result.
+    /// </summary>
+    public static Result<TEntity> Success => new(default, default, default);
```
Comment (Nihlus) [inline Remora.Results/Result.cs]: See previous comment. You can use the typed FromSuccess method as well.

**Review by Nihlus — CHANGES_REQUESTED** [review]

File: Remora.Results/Result.cs (hunk 4 above)
Comment (MikyM) [reply Remora.Results/Result.cs]: Won't accessing a `Result Success { get; } = ...` property return a copy of the struct anyway?

File: Remora.Results/Result.cs (hunk 5 above)
Comment (MikyM) [reply Remora.Results/Result.cs]: This would require suppressing a possible null ref warning as `Result<TEntity>.FromSuccess`' entity arg isn't marked as nullable.

File: Remora.Results/Result.cs (hunk 4 above)
Comment (Nihlus) [reply Remora.Results/Result.cs]: A copy, yes, but a copy is not the same as a new instance. The compiler can make other optimization decisions when doing copies, while creating an entirely new object has additional requirements.

File: Remora.Results/Result.cs (hunk 5 above)
Comment (Nihlus) [reply Remora.Results/Result.cs]: Hm, odd. I'll have to look into that. As it is, you can ignore that suggestion.

File: Remora.Results/Result.cs (hunk 4 above)
Comment (MikyM) [reply Remora.Results/Result.cs]: Fixed.
